src/moonlink_service/src/rest_api.rs

```diff
@@ -436,6 +453,35 @@ async fn upload_files(
     Ok(Json(FileUploadResponse { lsn }))
 }
 
+async fn optimize_table(
+    Path(table): Path<String>,
+    State(state): State<ApiState>,
+    Json(payload): Json<OptimizeTableRequest>,
+) -> Result<Json<OptimizeTableResponse>, (StatusCode, Json<ErrorResponse>)> {
+    debug!(
+        "Received table optimize request for '{}': {:?}",
+        table, payload
+    );
+    match state
+        .backend
+        .optimize_table(payload.database, payload.table, payload.mode.as_str())
+        .await
+    {
+        Ok(_) => Ok(Json(OptimizeTableResponse {})),
+        Err(e) => {
+            error!("Failed to optimize table '{}': {}", table, e);
```

Comment:
A note to myself, I need to revisit whether error logging here is good and needed.